pkg/operator/ceph/cluster/mon/service.go

```diff
@@ -67,6 +69,38 @@ func (c *Cluster) createService(mon *monConfig) (string, error) {
 		return "", nil
 	}
 
-	logger.Infof("mon %q ip is %s", mon.DaemonName, s.Spec.ClusterIP)
+	if c.spec.Network.MultiClusterService.Enabled {
```

Comment:
if we move this check to the `health.go`, it would mean that mons will always start with `clusterIP` for the first time and then failover to use `externalIP`.

Reply:
Updated the PR. Current code exports the service and deletes the mon canary pod at following: 
1. `initMonIPs`:  When the mons are first created with `multlcusterService.Enabled` true. 
2. During health check: If multiclusterService is enabled, check if the service is exported or if the exportedIP matches the mon IP.  If not, failover.  In the failover, export the service and delete the mon canary pod.